### PR TITLE
Remove sending profiles

### PR DESCRIPTION
**Pull request by jeyrey**

### CODE REVIEW

File: models/smtp.go [hunk 1]
```diff
@@ -5,190 +5,70 @@ import (
...
-// in the SMTP configuration
-var ErrHostNotSpecified = errors.New("No SMTP Host specified")
-
-// ErrInvalidHost indicates that the SMTP server string is invalid
-var ErrInvalidHost = errors.New("Invalid SMTP server address")
-
-// TableName specifies the database tablename for Gorm to use
-func (s SMTP) TableName() string {
-	return "smtp"
+func (this *SMTPRelayer) Init() error {
```
Comment (zzma) [inline models/smtp.go]: Go convention is to use a more descriptive variable name here - `this` is not usually used. See: https://stackoverflow.com/questions/29028512/go-this-keyword

File: worker/worker.go [hunk 2]
```diff
@@ -342,7 +342,7 @@ func SendTestEmail(s *models.SendTestEmailRequest) error {
 		}(a))
 	}
 
-	Logger.Printf("Sending Email to %s\n", s.Email)
+	Logger.Printf("Sending Test Email to %s\n", s.Target)
```
Comment (zzma) [inline worker/worker.go]: Why this change?

**Comment by zzma** [comment]
Two additional things: 

1.  add a database migration that removes the `smtp` table - there should be an example for removing the `pages` table. 
2. change Robert M Carr references to K Randolph in the email template

File: worker/worker.go (hunk 2 above)
Comment (zzma) [reply worker/worker.go]: Looks like it's causing a test failure `worker/worker.go:345: Printf format %s has arg s.Target of wrong type github.com/teamnsrg/gophish/models.Target`

File: gophish.go [hunk 3]
```diff
@@ -83,6 +83,7 @@ func main() {
 	// Load the config
 	config.LoadConfig(*configPath)
 	config.Version = string(version)
+
```
[the remark this answers is not in the document]
Comment (jeyrey) [reply gophish.go]: It breaks up a dense bit of code :)

File: models/smtp.go (hunk 1 above)
Comment (jeyrey) [reply models/smtp.go]: That is the most descriptive variable name possible. This is a member function operating on its parent object, this.

File: worker/worker.go (hunk 2 above)
Comment (jeyrey) [reply worker/worker.go]: Good catch

File: models/smtp.go (hunk 1 above)
Comment (jeyrey) [reply models/smtp.go]: What would you call it to be more descriptive?

Comment (zzma) [reply models/smtp.go]: `relayer`

Comment (jeyrey) [reply models/smtp.go]: then how will you know it is the parent object?

**Comment by jeyrey** [comment]
image

File: config/config.go [hunk 4]
```diff
@@ -24,16 +24,35 @@ type PhishServer struct {
...
 
+// Custom Headers for SMTP Relay
+type HeaderConf struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+// SMTP Relay config
+type SMTPRelayConf struct {
+	Interface        string       `json:"interface_type"`
```
Comment (zzma) [inline config/config.go]: This seems a little redundant if we are naming the struct `SMTPRelayConf` - I'd suggest removing this field or renaming the entire struct to `EmailRelayConf`

**Comment by jeyrey** [comment]
We can update it with more accurate content before we send it off.

File: models/smtp.go (hunk 1 above)
Comment (zzma) [reply models/smtp.go]: It's not really that important that it is the parent object, since go emphasizes considering everything as a function over methods. There are no private fields in a struct that only methods can access.

File: controllers/api.go [hunk 5]
```diff
@@ -284,81 +284,23 @@ func API_Groups_Id_Summary(w http.ResponseWriter, r *http.Request) {
...
-		}
-		s.ModifiedDate = time.Now().UTC()
-		s.UserId = ctx.Get(r, "user_id").(int64)
-		err = models.PostSMTP(&s)
-		if err != nil {
-			JSONResponse(w, models.Response{Success: false, Message: err.Error()}, http.StatusInternalServerError)
-			return
-		}
-		JSONResponse(w, s, http.StatusCreated)
+		JSONResponse(w, nil, http.StatusMethodNotAllowed)
```
Comment (zzma) [inline controllers/api.go]: we can make this the default case, which will handle all the other HTTP verbs as well

File: controllers/api.go [hunk 6]
```diff
@@ -284,81 +284,23 @@ func API_Groups_Id_Summary(w http.ResponseWriter, r *http.Request) {
...
-	s, err := models.GetSMTP(id)
-	if err != nil {
-		JSONResponse(w, models.Response{Success: false, Message: "SMTP not found"}, http.StatusNotFound)
-		return
-	}
+	s := models.SMTP
 	switch {
 	case r.Method == "GET":
 		JSONResponse(w, s, http.StatusOK)
 	case r.Method == "DELETE":
```
Comment (zzma) [inline controllers/api.go]: same here - make default case

File: models/smtp.go [hunk 7]
```diff
@@ -5,190 +5,70 @@ import (
...
 // Validate ensures that SMTP configs/connections are valid
-func (s *SMTP) Validate() error {
+func (this *SMTPRelayer) validate() error {
 	switch {
-	case s.FromAddress == "":
-		return ErrFromAddressNotSpecified
-	case s.Host == "":
-		return ErrHostNotSpecified
+	case this.FromAddress == "":
+		return errors.New("No MAIL FROM address specified")
```
Comment (zzma) [inline models/smtp.go]: Why replace these named errors?
